### PR TITLE
rerun ADaM templates

### PR DESCRIPTION
**Pull request by kaz462**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: data-raw/create_adams_data.R
Comment (kaz462) [inline data-raw/create_adams_data.R:1]: @dgrassellyb Tagging you for review here as this file is updated - need to rerun the ADaM templates from the latest main branch. Thank you!

File: NEWS.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## New Features
 
  - Variables labels are applied to datasets by `metacore`/`metatools`.
+ - ADPC, ADPPK and ADPP are updated to handle urine records.
```
Comment (manciniedoardo) [inline NEWS.md:6]: Shall we start adding issue numbers here? Maybe can do so for the other bullet pts as well.

**Review by manciniedoardo — CHANGES_REQUESTED** [review]

File: data-raw/create_adams_data.R [hunk 2]
```diff
@@ -122,18 +122,18 @@ run_template <- function(tp) {
 
 
 if (update_pkg) {
-  github_pat <- Sys.getenv("GITHUB_TOKEN") # in case of run through github workflows
```
Comment (dgrassellyb) [inline data-raw/create_adams_data.R:125]: I would uncomment this, it's important to use access token to avoid github access rate issues

File: data-raw/create_adams_data.R [hunk 3]
```diff
@@ -152,7 +152,7 @@ for (pkg in packages_list) {
   if (update_pkg) {
     # TODO: replace by main once done
     remotes::install_github(sprintf("pharmaverse/%s", pkg),
-      ref = "workflow_dispatch_templates", auth_token = github_pat,
```
Comment (dgrassellyb) [inline data-raw/create_adams_data.R:155]: same as previous comment I would put back `auth_token = github_pat`

File: data-raw/create_adams_data.R
Comment (dgrassellyb) [reply data-raw/create_adams_data.R:1]: Hi @kaz462 I just put some comments about access token, but except this little issue, everything looks fine to me !

File: NEWS.md (hunk 1 above)
Comment (kaz462) [reply NEWS.md]: good point, updated

File: data-raw/create_adams_data.R [hunk 4]
```diff
@@ -122,18 +122,18 @@ run_template <- function(tp) {
...
-  # if (github_pat == "") {
-  #   token_file <- "data-raw/github_token.R"
-  #   if (file.exists(token_file)) {
-  #     source(token_file)
-  #   } else {
-  #     "WARNING - you might have issues installing pharmaverse deps -
-  #           Please follow instructions in the readme.md"
-  #   }
-  # }
+  github_pat <- Sys.getenv("GITHUB_TOKEN") # in case of run through github workflows
```
Comment (dgrassellyb) [inline data-raw/create_adams_data.R]: @kaz462 since we made `GITHUB_TOKEN` optional  (line 155) I can we can remove this part : 
```
  if (github_pat == "") {
    token_file <- "data-raw/github_token.R"
    if (file.exists(token_file)) {
      source(token_file)
    } else {
      "WARNING - you might have issues installing pharmaverse deps -
            Please follow instructions in the readme.md"
    }
  }
  ```

Comment (kaz462) [reply data-raw/create_adams_data.R:125]: removed. Any idea why the workflows have failed?

**Comment by shajoezhu** [comment]
i was wondering if you guys would like to use staged.dependencies for these development? https://github.com/openpharma/staged.dependencies @kaz462 @manciniedoardo @dgrassellyb

File: data-raw/create_adams_data.R (hunk 4 above)
Comment (dgrassellyb) [reply data-raw/create_adams_data.R:125]: yes global issue from admiralci, @cicdguy pushed a fix actually :) https://github.com/pharmaverse/admiralci/pull/190/files 
So we might just need to rerun once it's merged

File: data-raw/create_adams_data.R [hunk 5]
```diff
@@ -123,17 +123,8 @@ run_template <- function(tp) {
...
-    if (file.exists(token_file)) {
-      source(token_file)
-    } else {
-      "WARNING - you might have issues installing pharmaverse deps -
-            Please follow instructions in the readme.md"
-    }
-  }
   # install pharmaversesdtm dep: TODO: see if we install from github or latest release?
-  install.packages("pharmaverse/pharmaversesdtm")
+  remotes::install_github("pharmaverse/pharmaversesdtm", ref = "main", force = TRUE)
```
[the remark this answers is not in the document]
Comment (kaz462) [reply data-raw/create_adams_data.R]: It's good now, @cicdguy @dgrassellyb thanks!

**Comment by manciniedoardo** [comment]
Update from 2024-05-16 meeting: we will do a re run on 31st May and then merge, unless meeting with NEST team next week results in further requests for changes.

File: R/adae.R [hunk 6]
```diff
@@ -1,5 +1,6 @@
-#' Adverse Event Analysis
+#' Dataset adae
 #'
+#' Adverse Events Analysis
```
Comment (sadchla-codes) [inline R/adae.R:3]: I think the label for each of the datasets (using this one as an example) have been reversed to the old one. Is there a step we missed

File: R/adrs_onco.R [hunk 7]
```diff
@@ -69,9 +70,12 @@
 #'     \item{ EOSSTT }{End of Study Status}
 #'     \item{ FRVDT }{Final Retrievel Visit Date}
 #'     \item{ DTHDT }{Date of Death}
-#'     \item{ DTHDTF }{undocumented field}
+#'     \item{ DTHDTF }{Date of Death Imputation Flag}
 #'     \item{ DTHADY }{Relative Day of Death}
 #'     \item{ LDDTHELD }{Elapsed Days from Last Dose to Death}
+#'     \item{ DTHCAUS }{undocumented field}
+#'     \item{ DTHDOM }{undocumented field}
+#'     \item{ DTHCGR1 }{undocumented field}
```
Comment (sadchla-codes) [inline R/adrs_onco.R:78]: These variables are missing their label. Should that update be made directly in the R/adrs_onco.R file, or the spec?

File: R/adtr_onco.R [hunk 8]
```diff
@@ -89,9 +90,12 @@
 #'     \item{ EOSSTT }{End of Study Status}
 #'     \item{ FRVDT }{Final Retrievel Visit Date}
 #'     \item{ DTHDT }{Date of Death}
-#'     \item{ DTHDTF }{undocumented field}
+#'     \item{ DTHDTF }{Date of Death Imputation Flag}
 #'     \item{ DTHADY }{Relative Day of Death}
 #'     \item{ LDDTHELD }{Elapsed Days from Last Dose to Death}
+#'     \item{ DTHCAUS }{undocumented field}
+#'     \item{ DTHDOM }{undocumented field}
+#'     \item{ DTHCGR1 }{undocumented field}
```
Comment (sadchla-codes) [inline R/adtr_onco.R:98]: Same issue as R/adrs_onco.R

**Review by sadchla-codes — CHANGES_REQUESTED** [review]

File: R/adae.R (hunk 6 above)
Comment (kaz462) [reply R/adae.R:3]: I applied label for the description only in  [create_adams_data.R](https://github.com/pharmaverse/pharmaverseadam/blob/5304a9143dc338cfa66c1094f88cd6f90e877ccb/data-raw/create_adams_data.R#L37-L39R)- 
image
vs

https://pharmaverse.github.io/pharmaverseadam/reference/adae.html has labels for both title and description 
image


Let me know your preference, I can do a quick update if needed :)

File: R/adrs_onco.R (hunk 7 above)
Comment (kaz462) [reply R/adrs_onco.R:78]: Nice catch! the update should be made directly in the `inst/extdata/adams-specs.xlsx`

File: R/adae.R (hunk 6 above)
Comment (sadchla-codes) [reply R/adae.R:3]: I think the first one is better without 'Dataset'
Screenshot 2024-05-29 at 1 59 56 PM

File: R/adrs_onco.R (hunk 7 above)
Comment (kaz462) [reply R/adrs_onco.R]: updated

**Review by sadchla-codes — APPROVED** [review]
looks great to me!

**Comment by shajoezhu** [comment]
hey guys, we have checked and everything looks in order. I think this is ready to be merged. Thanks so much! @kaz462 @manciniedoardo

**Review by manciniedoardo — APPROVED** [review]
